Install gRPC et al for testing except py26/lint/docs-rtd

tseaver opened the pull request:
Rationalize tox vs. gRPC:

- Install `grpcio` and related deps by default, and omit adding `_testing` to PYTHONPATH.

- For Python 2.6, and lint, RTD, suppress installing gRPC-related deps, and add `_testing` to PYTHONPATH.

Bump minimum version for `grpcio` to 0.15.0.

Fix up Py3k failures with `grpcio` installed.

@daspecster, @dhermes Maybe this should land before 0.17.0?

tseaver commented:
I'm going to go ahead with 0.17.0 without this one.

dhermes commented:
Changes look mostly good. Failures caused by what?

tseaver commented:
Ugh, no idea -- I can't reproduce on my Ubuntu Trusty machine here. "Two hard things in computer science:  caching, naming things, and off-by-one errors."

daspecster commented:
@tseaver it did the same thing to me on OSX, someone reported it here and I opened an issue on grpc.

I believe grpc needs to do a release before this works again.
https://github.com/grpc/grpc/issues/7135#issuecomment-229456308

How I reproduce the issue.

I made a virtualenv with py34.
```bash
mkvirtualenv -p /usr/local/bin/python3.4 test-multi
python setup.py install
pip install tox
tox -e py34
```
Key points from the log output that it mentions in the error...
```
Command "/Users/me/Documents/google/gcloud-python/.tox/py34/bin/python3.4 -u -c "import setuptools, 
tokenize;__file__='/private/var/folders/bc/htqxjjns6gv4g1kspb2zppm80000gn/T/pip-build-
bifiqtz2/grpcio/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), 
__file__, 'exec'))" install --record /var/folders/bc/htqxjjns6gv4g1kspb2zppm80000gn/T/pip-ndumdegl-
record/install-record.txt --single-version-externally-managed --compile --install-headers 
/Users/me/Documents/google/gcloud-python/.tox/py34/bin/../include/site/python3.4/grpcio" failed with 
error code 1 in /private/var/folders/bc/htqxjjns6gv4g1kspb2zppm80000gn/T/pip-build-bifiqtz2/grpcio/
```

daspecster commented:
@tseaver, should we just close this for now and hope that future `grpc` fixes it?

tseaver commented:
@daspecster 

> Should we just close this for now and hope that future grpc fixes it?

I think we want to be where this PR is (the rationalize `tox.ini`, particularly).  We need to have `grpcio` wheels published for Python 3.4+ (including Windows) before our story improves.

tseaver commented:
@dhermes, @daspecster I rebased to fix conflicts, and also updated the version pins for the gRPC-related dependencies to reflect today's grpcio-1.0rc1 release, as well as comments from @bjwatson on potential future breaking changes for the GAX libraries.

tseaver commented:
Woohoo!  Travis is happy, at least for the PR.

daspecster commented on tox.ini:
```diff
@@ -23,29 +19,37 @@ covercmd =
+    {[testing]deps}
     {[grpc]deps}
+
+[testenv:py26]
```
@tseaver, py26 isn't mentioned in the envlist on line 3? Also, py27 is gone and only py27-pandas is available?

tseaver replied:
@daspecster We already dropped `py26` as a "run-by-default" environment:  this environment is for when we run `tox -e py26` by hand, or from Travis.  The `py27` environment now uses the defaults from `[testenv]`.

daspecster replied:
Ok, I was confused on how envlist actually worked with the tox default environments. Thanks @tseaver!

tseaver commented:
@dhermes, @daspecster any remaining issues?

dhermes commented on tox.ini:
```diff
@@ -23,29 +19,37 @@ covercmd =
-    python2.7
+[docs]
+deps =
+    {[testenv]deps}
```
Shouldn't this be `[testing]`?

dhermes replied:
NVM

dhermes commented:
LGTM

bjwatson commented on setup.py:
```diff
@@ -20,10 +20,12 @@
+    'gax-google-pubsub-v1 >= 0.7.12, < 0.8dev',
+    'grpc-google-pubsub-v1 >= 0.7.12, < 0.8dev',
+    'gax-google-logging-v2 >= 0.7.12, < 0.8dev',
+    'grpc-google-logging-v2 >= 0.7.12, < 0.8dev',
```
What does the "dev" string do?

daspecster replied:
It's a PEP440 standard I believe. https://packaging.python.org/distributing/#standards-compliance-for-interoperability

tseaver replied:
It indicates the earliest possible pre-release of the `0.8` series.
